### PR TITLE
Proofpoint Feed Integration Fix for firstseenbysource and lastseenbysource fields

### PR DESCRIPTION
**Pull request by kbajkowski**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
Currently, the code in the integration attempts to set XSOAR fields firstseenbysource and lastseenbysource with a date string that XSOAR does not like. This can result in thousands of error lines being printed to the server.log file depending on how many indicators are being fetched, and also it causes those two fields to always be set to N/A in XSOAR.

The code I'm introducing will attempt to format the first_seen and last_seen variables using the dateparser's isoformat() function. If for whatever reason those variables are "None" (i.e: Proofpoint sent them into XSOAR as "None") then it'll just set those variables to an empty string and not attempt to run isoformat() to avoid issues.

I have tested this in my client's environment and it got rid of the error messages in our server.log, while also populating the first seen by source & last seen by source fields. 

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Comment by anas-yousef** [comment]
@kbajkowski Thank you for your contribution!
Please bump the pack version in the pack_metadata.json file and add notes describing your changes to a new Release Notes file.
To do that, please run the demisto-sdk update-release-notes command - for more information please see: https://xsoar.pan.dev/docs/concepts/demisto-sdk#update-release-notes.

Can you send me an example of the values returned from **item.get("first_seen","")** and **item.get("last_seen","")**, so I can understand better why the system does not support these values? What errors does the customer see in their server logs?

File: Packs/FeedProofpoint/Integrations/FeedProofpoint/FeedProofpoint.py [hunk 1]
```diff
@@ -109,15 +110,34 @@ def _build_iterator(
...
+        else:
+            first_seen = ""
+
+        if last_seen_raw:
+            try:
+                last_seen = dateparser.parse(last_seen_raw).isoformat()
+            except AttributeError:
+                last_seen = ""
+        else:
+            last_seen = ""
```
Comment (anas-yousef) [inline Packs/FeedProofpoint/Integrations/FeedProofpoint/FeedProofpoint.py:130]: Consider creating a function that takes in a value of type string, and tries to run **dateparser.parse(value).isoformat()** on it and returns, and if there is an **AttributeError** exception, then return an empty string, then use that function on both **item.get("first_seen","")** and **item.get("last_seen","")**, resulting in less code duplication.

**Comment by kbajkowski** [comment]
Hey @anas-yousef, thanks for the suggestions, I could certainly take a stab at creating a function.

Here is what the server.log has when the integration is ingesting indicator and trying to set the firstseenbysource and lastseenbysource fields. XSOAR is expecting an [ISO8601](https://www.iso.org/iso-8601-date-and-time-format.html) timestamp because the field it is trying to write to is a "Date Picker" field and not just a short/long text field.

2024-05-02 18:08:25.2613 error Failed to convert [2024-04-21] to type [date] from custom fields args [error ''2024-04-21' is not a legal date, field: lastseenbysource (8902)'] (source: /builds/GOPATH/src/gitlab.xdr.pan.local/xdr/xsoar/server/domain/incident.go:702) 

2024-05-02 18:08:25.2616 error Failed to convert [2021-01-19] to type [date] from custom fields args [error ''2021-01-19' is not a legal date, field: firstseenbysource (8902)'] (source: /builds/GOPATH/src/gitlab.xdr.pan.local/xdr/xsoar/server/domain/incident.go:702) 

2024-05-02 18:08:25.2616 error Failed to convert [2024-04-20] to type [date] from custom fields args [error ''2024-04-20' is not a legal date, field: lastseenbysource (8902)'] (source: /builds/GOPATH/src/gitlab.xdr.pan.local/xdr/xsoar/server/domain/incident.go:702) 

2024-05-02 18:08:25.2617 error Failed to convert [2024-01-11] to type [date] from custom fields args [error ''2024-01-11' is not a legal date, field: firstseenbysource (8902)'] (source: /builds/GOPATH/src/gitlab.xdr.pan.local/xdr/xsoar/server/domain/incident.go:702) 

2024-05-02 18:08:25.2617 error Failed to convert [2024-05-02] to type [date] from custom fields args [error ''2024-05-02' is not a legal date, field: lastseenbysource (8902)'] (source: /builds/GOPATH/src/gitlab.xdr.pan.local/xdr/xsoar/server/domain/incident.go:702) 

2024-05-02 18:08:25.2618 error Failed to convert [2024-04-06] to type [date] from custom fields args [error ''2024-04-06' is not a legal date, field: firstseenbysource (8902)'] (source: /builds/GOPATH/src/gitlab.xdr.pan.local/xdr/xsoar/server/domain/incident.go:702)

**Comment by anas-yousef** [comment]
Hey @kbajkowski let us take this over Slack, can you ping me?

**Comment by anas-yousef** [comment]
Hi @kbajkowski, 
Thank you for your contribution. 
I have not received your response in the last two weeks, so I will close your PR now.
Please feel free to re-open it whenever you're available. 
Thanks again.

**Pull request closed without merging**